components/brave_wallet/browser/asset_discovery_manager.h

```diff
@@ -0,0 +1,113 @@
+  void KeyringCreated(const std::string& keyring_id) override {}
+  void KeyringRestored(const std::string& keyring_id) override {}
+  void KeyringReset() override {}
+  void Locked() override {}
+  void Unlocked() override {}
+  void BackedUp() override {}
+  void AccountsChanged() override {}
+  void AccountsAdded(mojom::CoinType coin,
+                     const std::vector<std::string>& addresses) override;
+  void AutoLockMinutesChanged() override {}
+  void SelectedAccountChanged(mojom::CoinType coin) override {}
+
+  using APIRequestResult = api_request_helper::APIRequestResult;
+  using EthGetLogsCallback =
+      base::OnceCallback<void(const std::vector<Log>& logs,
+                              mojom::ProviderError error,
+                              const std::string& error_message)>;
+  void DiscoverAssets(const std::string& chain_id,
+                      mojom::CoinType coin,
+                      const std::vector<std::string>& account_addresses,
+                      bool update_prefs,
+                      const std::string& from_block,
+                      const std::string& to_block,
+                      const std::vector<std::string>& supported_chain_ids);
+
+  void OnGetAllTokensDiscoverAssets(
+      const std::string& chain_id,
+      const std::vector<std::string>& account_addresses,
+      std::vector<mojom::BlockchainTokenPtr> user_assets,
+      bool update_prefs,
+      const std::string& from_block,
+      const std::string& to_block,
+      std::vector<mojom::BlockchainTokenPtr> token_list);
+
+  void OnGetTransferLogs(
+      base::flat_map<std::string, mojom::BlockchainTokenPtr>& tokens_to_search,
+      bool update_next_asset_discovery_from_block,
+      const std::string& chain_id,
+      const std::vector<Log>& logs,
+      mojom::ProviderError error,
+      const std::string& error_message);
+
+  void CompleteDiscoverAssets(
+      const std::string& chain_id,
+      std::vector<mojom::BlockchainTokenPtr> discovered_assets,
+      mojom::ProviderError error,
+      const std::string& error_message);
+
+  // Triggered by when KeyringService emits AccountsAdded event.
+  // Rate limits will be ignored, and eth_getLogs query
+  // will run against all blocks, "earliest" to "latest".
+  void DiscoverAssetsOnAllSupportedChains(
+      const std::vector<std::string>& account_addresses,
+      const std::vector<std::string>& supported_chain_ids =
+          GetAssetDiscoverySupportedChains());
+
+  // Called by frontend via BraveWalletService.
+  // Subject to client side rate limiting based on
+  // kBraveWalletLastDiscoveredAssetsAt pref value. Only runs eth_getLogs
+  // against block range between
+  // kBraveWalletNextAssetDiscoveryFromBlocks pref and "latest".
+  void DiscoverAssetsOnAllSupportedChainsOnRefresh(
+      const std::vector<std::string>& account_addresses,
+      const std::vector<std::string>& supported_chain_ids =
+          GetAssetDiscoverySupportedChains());
```

Comment:
Why are they all public, please only make functions that we're going to expose to other classes public.